objectivec/cxx_utils.mm

```diff
@@ -54,23 +54,20 @@
 NSArray<NSString*>* toNSStringNSArray(const std::vector<std::string>& strs) {
   NSMutableArray<NSString*>* result = [NSMutableArray arrayWithCapacity:strs.size()];
   for (const std::string& str : strs) {
-    NSString* nsStr = [NSString stringWithUTF8String:str.c_str()];
-    if (nsStr) {
-      [result addObject:nsStr];
-    } else {
-      ORT_CXX_API_THROW("Failed to convert std::string to NSString", ORT_INVALID_ARGUMENT);
-    }
+    [result addObject:toNSString(str)];
   }
   return result;
 }
 
 NSArray<ORTValue*>* _Nullable wrapUnownedCAPIOrtValues(const std::vector<OrtValue*>& values, NSError** error) {
   NSMutableArray<ORTValue*>* result = [NSMutableArray arrayWithCapacity:values.size()];
   for (size_t i = 0; i < values.size(); ++i) {
-    ORTValue* val = [[ORTValue alloc] initWithCAPIOrtValue:values[i] externalTensorData:nil error:error];
+    ORTValue* val = [[ORTValue alloc] initWithCXXAPIOrtValue:Ort::Value{values[i]}
+                                          externalTensorData:nil
+                                                       error:error];
     if (!val) {
       // clean up all the C API Ortvalues which haven't been wrapped by ORTValue
-      for (size_t j = i; j < values.size(); ++j) {
+      for (size_t j = i + 1; j < values.size(); ++j) {
         Ort::GetApi().ReleaseValue(values[j]);
```

Comment:
Should we also use Ort::Value here for the release so we're more consistent?

```suggestion
        Ort::Value(values[j]);  // construct a temporary Ort::Value who's dtor will handle the ReleaseValue call.
```

Reply:
it's an option, but IMO that seems more indirect and worth adding the comment about. if you prefer that I can update.

Reply:
If we could add a comment for the call to initWithCXXAPIOrtValue so the usage of the std::move'd temporary Ort::Value there is explained I think that would help future developers as it requires a fair bit of intrinsic knowledge about c++ and the implementations to fully understand.

Reply:
added a comment and tried to make it clearer. maybe it was doing too much in one line.